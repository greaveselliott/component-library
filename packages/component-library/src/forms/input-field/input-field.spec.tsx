import React from 'react';
import { shallow } from 'enzyme';

import InputField from './input-field.component';

test('should render', () =>
  expect(
    shallow(<InputField id="abc" name="abc" placeholder="abc" />),
  ).toMatchSnapshot());

test('should render with password type', () =>
  expect(
    shallow(
      <InputField id="abc" name="abc" placeholder="abc" type="password" />,
    ),
  ).toMatchSnapshot());

test('should render with email type', () =>
  expect(
    shallow(<InputField id="abc" name="abc" placeholder="abc" type="email" />),
  ).toMatchSnapshot());

test('should render with an icon', () =>
  expect(
    shallow(
      <InputField
        id="abc"
        name="abc"
        placeholder="abc"
        Icon={() => <div>Fancy icon</div>}
      />,
    ),
  ).toMatchSnapshot());
