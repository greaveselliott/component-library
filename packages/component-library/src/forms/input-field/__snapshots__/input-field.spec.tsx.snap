// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="input-field"
>
  <input
    className="input-field__field"
    id="abc"
    name="abc"
    placeholder="abc"
    type="text"
  />
</div>
`;

exports[`should render with an icon 1`] = `
<div
  className="input-field"
>
  <div
    className="input-field__icon"
  >
    <Icon />
  </div>
  <input
    className="input-field__field input-field__field--with-icon"
    id="abc"
    name="abc"
    placeholder="abc"
    type="text"
  />
</div>
`;

exports[`should render with email type 1`] = `
<div
  className="input-field"
>
  <input
    className="input-field__field"
    id="abc"
    name="abc"
    placeholder="abc"
    type="email"
  />
</div>
`;

exports[`should render with password type 1`] = `
<div
  className="input-field"
>
  <input
    className="input-field__field"
    id="abc"
    name="abc"
    placeholder="abc"
    type="password"
  />
</div>
`;
